src/brsTypes/components/RoSGNode.ts

```diff
@@ -213,6 +230,129 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
             return this.get(new BrsString(lKey));
         },
     });
+
+    /* Return the current number of children in the subject node list of children.
+    This is always a non-negative number. */
+    private getchildcount = new Callable("getchildcount", {
+        signature: {
+            args: [],
+            returns: ValueKind.Int32,
+        },
+        impl: (interpreter: Interpreter) => {
+            return new Int32(this.children.length);
+        },
+    });
+
+    /* Adds a child node to the end of the subject node list of children so that it is
+    traversed last (of those children) during render. */
+    private appendchild = new Callable("appendchild", {
+        signature: {
+            args: [new StdlibArgument("child", ValueKind.Dynamic)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, child: BrsType) => {
+            for (let childNode of this.children) {
+                if (childNode === child) {
+                    return BrsBoolean.True;
+                }
+            }
+            if (child instanceof RoSGNode) {
+                this.children.push(child);
+                child.setParent(this);
+                return BrsBoolean.True;
+            }
+            return BrsBoolean.False;
+        },
+    });
+
+    /* Retrieves the number of child nodes specified by num_children from the subject
+    node, starting at the position specified by index. Returns an array of the child nodes
+    retrieved. If num_children is -1, return all the children. */
+    private getchildren = new Callable("getchildren", {
+        signature: {
+            args: [
+                new StdlibArgument("num_children", ValueKind.Int32),
+                new StdlibArgument("index", ValueKind.Int32),
+            ],
+            returns: ValueKind.Object,
+        },
+        impl: (interpreter: Interpreter, num_children: Int32, index: Int32) => {
+            let numChildrenValue = num_children.getValue();
+            let indexValue = index.getValue();
+            let childrenSize = this.children.length;
+            if (numChildrenValue <= -1 && indexValue === 0) {
+                //short hand to return all children
+                return new RoArray(this.children.slice());
+            } else if (numChildrenValue <= 0 || indexValue < 0 || indexValue >= childrenSize) {
+                //these never return any children
+                return new RoArray([]);
+            } else {
+                //only valid cases
+                let endSliceIndex = Math.min(indexValue + numChildrenValue, childrenSize);
```

Comment:
Looks like this `Math.min` isn't necessary – Javascript's `Array.prototype.slice()` won't attempt to read past the end of an array, and effectively does this `min` operation internally!